test/units/testsuite-60.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+set -eux
+set -o pipefail
+
+systemd-analyze log-level debug
+systemd-analyze log-target journal
+
+NUM_DIRS=20
+
+# mount/unmount enough times to trigger the /proc/self/mountinfo parsing rate limiting
+
+for ((i = 0; i < NUM_DIRS; i++)); do
+    mkdir "/tmp/meow$i"
+done
+
+for ((i = 0; i < NUM_DIRS; i++)); do
+    mount -t tmpfs tmpfs "/tmp/meow$i"
+done
+
+systemctl daemon-reload
+systemctl list-units -t mount tmp-meow* | grep tmp-meow
+
+for ((i = 0; i < NUM_DIRS; i++)); do
+    umount "/tmp/meow$i"
+done
+
+# verify that we successfully entered and exited the rate limit state
+
+timeout="$(date -ud "2 minutes" +%s)"
+while [[ $(date -u +%s) -le $timeout ]]; do
+    if journalctl -u init.scope | grep "(mount-monitor-dispatch) entered rate limit"; then
+        break
+    fi
+    sleep 5
+done
+
+exited_rl=false
+timeout="$(date -ud "2 minutes" +%s)"
+while [[ $(date -u +%s) -le $timeout ]]; do
+    if journalctl -u init.scope | grep "(mount-monitor-dispatch) left rate limit"; then
+        exited_rl=true
+        break
+    fi
+    sleep 5
+done
+
+if ! "$exited_rl"; then exit 24; fi
```

Comment:
Maybe, it is safer to check this only when we have `"(mount-monitor-dispatch) entered rate limit"` in the above?
If the system is too slow to trigger the rate limit, then we will never see the `"left rate limit"` log message.

Reply:
I wasn't really sure what to do if for example the CI infra is too slow to trigger the rate limit. Maybe just skip the message checks but still verify the units are cleaned up at the end?

Reply:
I think we should check that the mount units are cleaned up at the end even if the system is too slow.